### PR TITLE
Misc small makefile fixes

### PR DESCRIPTION
**Pull request by mpg**
## Description

Small fixes to the Makefiles:
- typos in the windows portion;
- some generated files were incorrectly removed by `clean` - they're supposed to be only removed by `neat`.

Forward-ported from the [3.6 release candidate PR](https://github.com/Mbed-TLS/mbedtls-restricted/pull/1286).

## PR checklist

- [x] **changelog** not required because: small fixes to the Makefiles
- [x] **development PR** here
- [x] **framework PR** not required
- [x] **3.6 PR** not required because: already done as part of the release (landed with mergeback of the release branch)
- [x] **2.28 PR** not required because: 2.28 not affected
- **tests** not required because: Makefiles

### CODE REVIEW

**Review by tom-cosgrove-arm — DISMISSED** [review]
LGTM

File: tests/Makefile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-MBEDTLS_TEST_PATH = .
+mbEDTLS_TEST_PATH = .
```
Comment (davidhorstmann-arm) [inline tests/Makefile:1]: Hmm
This doesn't look right to me. Thankfully that part doesn't seem to be in the release candidate.

**Review by davidhorstmann-arm — CHANGES_REQUESTED** [review]

File: tests/Makefile (hunk 1 above)
Comment (mpg) [reply tests/Makefile:1]: Aw, I must have made a stupid mistake while resolving conflicts, thanks for catching it.

**Review by tom-cosgrove-arm — APPROVED** [review]
LGTM

File: tests/Makefile (hunk 1 above)
Comment (tom-cosgrove-arm) [reply tests/Makefile]: Bah, I read that change the wrong way round :(

**Review by davidhorstmann-arm — APPROVED** [review]
LGTM, thanks!

**Pull request merged**